### PR TITLE
Defect/618 add schema search path param

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: admin-service/src/test/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantValidationServiceTest.java [hunk 1]
```diff
@@ -321,19 +321,20 @@ private DataSourceUrlParts geturlParts(final String key, final String dataSource
...
-        DataSourceElementsTenant dataSource = new DataSourceElementsTenant();
-        dataSource.setUrlParts(geturlParts(key, dataSourceName));
+    private DataSourceElementsTenant dataSource(final String dataSourceName, final String key) {
+        final DataSourceElementsTenant dataSource = new DataSourceElementsTenant();
         dataSource.setUsername(key + "_username");
         // each olap datasource should get the same SchemaSearchPath
         if (dataSourceName.contains(RdwDatabaseType.REPORTING_OLAP.datasourceNameHint)) {
             dataSource.setSchemaSearchPath("redShiftPath");
+        } else {
+            dataSource.setUrlParts(geturlParts(key, dataSourceName));
```
Comment (esotrope) [inline admin-service/src/test/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantValidationServiceTest.java]: changes the expected minimal correct structure of the datasources.

File: webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.ts [hunk 2]
```diff
@@ -237,6 +237,11 @@ function setDefaultDatabaseNameAndUsername(
   const key = (inputKey || '').toLowerCase();
   const defaultUsername = key;
 
+  const archivePathPrefixControl = formGroup.controls[`archive.pathPrefix`];
+  if (archivePathPrefixControl.pristine) {
+    archivePathPrefixControl.patchValue(defaultUsername);
+  }
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.ts]: image

File: webapp/src/main/webapp/src/app/admin/tenant/model/field-configurations.ts [hunk 3]
```diff
@@ -46,70 +66,66 @@ export const fieldConfigurationsByKey: { [key: string]: FieldConfiguration } = {
...
+  'archive.s3RegionStatic': {
+    dataType: 'string'
   },
   'archive.s3SecretKey': {
     dataType: 'password'
   },
-  'archive.configuredFile': {
-    dataType: 'boolean'
+  'archive.s3sse': {
+    dataType: 'string'
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/model/field-configurations.ts]: image

File: webapp/src/main/webapp/src/app/admin/tenant/model/field.ts [hunk 4]
```diff
@@ -73,6 +74,11 @@ export interface FieldConfiguration<DataTypes = any, T = any> {
    * Used to provide leniency when comparing values for enumerations
    */
   equals?: Equals<T>;
+
+  /**
+   * If true the field will be omitted from the form
+   */
+  hidden?: (type: TenantType) => boolean;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/model/field.ts]: going to use this to hide things for sandbox

File: webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts [hunk 5]
```diff
@@ -36,6 +36,18 @@ function trimStrings(object: any): any {
   );
 }
 
+function omitByKey(object: any, matcher: (key: string) => boolean): any {
+  return transform(
+    object,
+    (result: any, value: any, key: string) => {
+      if (!matcher(key)) {
+        result[key] = value;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts]: used to omit default password and username values

File: webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts [hunk 6]
```diff
@@ -162,15 +165,21 @@ export function toConfigurations(
...
+          ignoreArraysOfPrimitives,
+          // collapse this field into one
+          ignoreKeys(key => key.startsWith('reporting.state'))
+        )
+      ),
+      // blank out any defaults for these values
+      key =>
+        /^(aggregate\.tenant|datasources\.\w+\.(username|password|urlParts\.database|schemaSearchPath))$/.test(
+          key
+        )
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts]: just don't take defaults for these things.

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantValidationService.java [hunk 7]
```diff
@@ -192,30 +192,31 @@ private void verifyTenantArchive(final TenantConfiguration tenantConfiguration,
...
-            if (dataSource.getValue().getUrlParts() == null) {
-                // default tenant information is not configured properly
-                String errorMsg = String.format("Cannot create Tenant: %s, key: %s, the Database property %s is not configure correctly." +
-                                " Please contact your admin.",
-                        tenantConfiguration.getTenant().getName(), tenantConfiguration.getTenant().getKey(),
-                        dataSource.getKey());
-                throw new IllegalArgumentException(errorMsg);
-            }
-            if (!isOlapDatasource(dataSource.getKey())) {
+            if (isOlapDatasource(dataSource.getKey())) {
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantValidationService.java]: thanks @esotrope  that looks right

**Pull request merged**